src/libxrpl/protocol/STTx.cpp

```diff
@@ -438,6 +475,77 @@ STTx::checkMultiSign(
     return {};
 }
 
+Expected<void, std::string>
+STTx::checkBatchMultiSign(
+    STObject const& batchSigner,
+    RequireFullyCanonicalSig requireCanonicalSig,
+    Rules const& rules) const
+{
+    Serializer msg;
+    serializeBatch(msg, getFlags(), getBatchTransactionIDs());
+    bool const fullyCanonical = (getFlags() & tfFullyCanonicalSig) ||
+        (requireCanonicalSig == RequireFullyCanonicalSig::yes);
+
+    return multiSignHelper(
+        batchSigner,
+        fullyCanonical,
+        [&msg](AccountID const&) -> std::vector<uint8_t> {
+            return msg.getData();
+        },
+        rules);
+}
+
+Expected<void, std::string>
+STTx::checkMultiSign(
+    RequireFullyCanonicalSig requireCanonicalSig,
+    Rules const& rules) const
+{
+    bool const fullyCanonical = (getFlags() & tfFullyCanonicalSig) ||
+        (requireCanonicalSig == RequireFullyCanonicalSig::yes);
+    return multiSignHelper(
+        *this,
+        fullyCanonical,
+        [this](AccountID const& accountID) -> std::vector<uint8_t> {
+            Serializer dataStart = startMultiSigningData(*this);
```

Comment:
Can just this line be moved out of the helper function, and passed in as a parameter? That might solve the computational load issue.

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/f1e293a687d6da2c256a41b527cfc6be24d84055